src/osdc/Objecter.cc

```diff
@@ -1208,18 +1208,29 @@ void Objecter::handle_osd_map(MOSDMap *m)
 
 	cluster_full = cluster_full || _osdmap_full_flag();
 	update_pool_full_map(pool_full_map);
+
+	// check all outstanding requests on every epoch, including need_resend
+	for (auto p = need_resend.begin(); p != need_resend.end(); ) {
+	  Op *op = p->second;
+	  ldout(cct, 10) << __func__ << "  checking op " << p->first << dendl;
+	  int r = _calc_target(&op->target, nullptr);
+	  if (r == RECALC_OP_TARGET_POOL_DNE) {
+	    p = need_resend.erase(p);
```

Comment:
op->session will have been set to NULL by _session_op_remove() when the op was added to need_resend...